ecmascript/transforms/module/src/common_js.rs

```diff
@@ -174,14 +177,71 @@ where
                     }
                     drop(scope);
                     drop(scope_ref_mut);
+
                     match item {
                         ModuleItem::ModuleDecl(ModuleDecl::ExportAll(export)) => {
                             self.scope
                                 .borrow_mut()
                                 .lazy_blacklist
                                 .insert(export.src.value.clone());
 
-                            export_alls.push(export);
+                            let mut scope_ref_mut = self.scope.borrow_mut();
+                            let scope = &mut *scope_ref_mut;
+
+                            if exported_names.is_none()
+                                && (export_alls.len() >= 1 || !exports.is_empty())
+                            {
+                                let exported_names_ident = private_ident!("_exportNames");
+                                stmts.push(ModuleItem::Stmt(Stmt::Decl(Decl::Var(VarDecl {
+                                    span: DUMMY_SP,
+                                    kind: VarDeclKind::Var,
+                                    decls: vec![VarDeclarator {
+                                        span: DUMMY_SP,
+                                        name: Pat::Ident(exported_names_ident.clone().into()),
+                                        init: Some(Box::new(Expr::Object(ObjectLit {
+                                            span: DUMMY_SP,
+                                            props: exports
+                                                .clone()
```

Comment:
Is this clone necessary?

Reply:
If I recall corretly yes, there was a move occurred somewhere.